### PR TITLE
Makes code more OS portable

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Almost ready for #105

### CODE REVIEW

File: internal/cmd/run_test.go [hunk 1]
```diff
@@ -95,7 +97,7 @@ func TestGetEnvoyRun_ReadsHomeVersionFile(t *testing.T) {
 	runWithoutConfig(t, c)
 
 	// No implicit lookup
-	require.NotContains(t, o.Out.(*bytes.Buffer).String(), "looking up latest version\n")
+	require.NotContains(t, o.Out.(*bytes.Buffer).String(), "looking up latest version"+lf)
```
Comment (mathetake) [inline internal/cmd/run_test.go:100]: I guess we could use `fmt.Sprintln` to produce a portable string with a trailing new line..

Comment (codefromthecrypt) [reply internal/cmd/run_test.go:100]: except it doesn't :) inside it does this

```
	p.buf.writeByte('\n')
```

Comment (codefromthecrypt) [reply internal/cmd/run_test.go:100]: it is very strange that go decided to not include portable linefeed support 🤷

Comment (mathetake) [reply internal/cmd/run_test.go:100]: funny 😂

Comment (codefromthecrypt) [reply internal/cmd/run_test.go:100]: that said, we could make our own Sprintln that does..

File: internal/cmd/run_test.go [hunk 2]
```diff
@@ -29,11 +28,14 @@ import (
 
 	rootcmd "github.com/tetratelabs/getenvoy/internal/cmd"
 	"github.com/tetratelabs/getenvoy/internal/globals"
+	"github.com/tetratelabs/getenvoy/internal/moreos"
 	"github.com/tetratelabs/getenvoy/internal/test"
 	"github.com/tetratelabs/getenvoy/internal/test/morerequire"
 	"github.com/tetratelabs/getenvoy/internal/version"
 )
 
+const lf = moreos.LineSeparator
```
Comment (mathetake) [inline internal/cmd/run_test.go:37]: what does "f" of lf mean here?

**Review by mathetake — APPROVED** [review]

File: internal/cmd/run_test.go (hunk 1 above)
Comment (codefromthecrypt) [reply internal/cmd/run_test.go:100]: meh... problem with doing this is that IDEs won't syntax highlight the format string

File: internal/cmd/run_test.go (hunk 2 above)
Comment (codefromthecrypt) [reply internal/cmd/run_test.go:37]: renamed to ls!

Comment (codefromthecrypt) [reply internal/cmd/run_test.go]: actually "ln" looks nicer as matches println

**Comment by codefromthecrypt** [comment]
ok also backfilled a ln test, though it will need a slight adjustment when completing windows it shows why it is there.

File: internal/moreos/moreos_test.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+	f, err := os.Stat(bin)
+	require.NoError(t, err)
+
+	require.False(t, isExecutable(f))
+}
+
+func TestLineSeparator(t *testing.T) {
+	stdout := new(bytes.Buffer)
+	cmd := exec.Command("echo", "cats")
```
Comment (codefromthecrypt) [inline internal/moreos/moreos_test.go:58]: in windows, we need to have an if statement and do `"cmd.exe", "/c", "echo cats"`. This proves that it works and helps ensure we catch newline things without running envoy

**Comment by codefromthecrypt** [comment]
Thanks for looking  and asking questions @mathetake !

**Pull request merged**